lib/front_desk.rb

```diff
@@ -0,0 +1,120 @@
+require_relative 'date_range'
+require_relative 'reservation'
+
+module Hotel
+  class FrontDesk
+    attr_accessor :reservations
+    
+    def initialize(reservations:nil )
+      @reservations = reservations || []
+    end
+
+    #access a list of all rooms in hotel
+    def list_all
+      display = ""
+      @reservations.map do |bookings| 
+        display += "Reservation_id = #{bookings.reservation_id} Start date = #{bookings.date_range.start_date} End_date =#{bookings.date_range.end_date} Room Number = #{bookings.room_num} Block Tag = #{bookings.block_tag}, Room Rate = #{bookings.room_rate}\n"
+      end 
+      return display
+    end 
+
+    #access a list of reservations for a specific room number and date range
+    def room_reservations_and_date(room_num,date_range)
+      display = ""
+      res_room_date_range = @reservations.select {|bookings| bookings.room_num == room_num && date_range.overlap(bookings.date_range)}
+      res_room_date_range.map do |bookings|
+        display += "Reservation_id = #{bookings.reservation_id} Start date = #{bookings.date_range.start_date} End_date =#{bookings.date_range.end_date} Room Number = #{bookings.room_num}\n"
+      end 
+      return display
+    end 
+
+    # check reservations of specific date
+    def check_date_reservations(date)
+      display = ""
+      reservations_include_a_date = @reservations.select {|bookings| bookings.date_range.include_date(date)==true}
+      reservations_include_a_date.map do |bookings|
+        display += "Reservation_id = #{bookings.reservation_id} Start date = #{bookings.date_range.start_date} End_date =#{bookings.date_range.end_date} Room Number = #{bookings.room_num}\n"
+      end 
+      return display
+    end 
+
+    #request reservation 
+    def request_reservation(start_date,end_date)
+      date_range = Hotel::DateRange.new(start_date:start_date,end_date:end_date)
+      times_overlap = @reservations.count{|bookings|bookings.date_range.overlap(date_range)}
+
+      available_rooms = self.room_available(date_range)
+      unless times_overlap < 20
+        raise ArgumentError, "cannot reserve room since all rooms are booked"
```

Comment:
This should be a custom error instead of an `ArgumentError` since nothing is wrong with `start_date` or `end_date`.

You can define it like:
```ruby
module Hotel
  class AllRoomsBookedError < StandardError
  end
end
```

And use it like:
```suggestion
        raise AllRoomsBookedError, "cannot reserve room since all rooms are booked"
```